podpac/core/node.py

```diff
@@ -385,12 +387,12 @@ def _base_definition(self):
             d["inputs"] = inputs
 
         # style
-        if self.style != Style() and self.style.definition:
+        if self.style.definition:
             d["style"] = self.style.definition
 
         return d
 
-    @cached_property
+    @property
```

Comment:
Why no longer a cached_property?

Reply:
We use a private attribute in cached_property, so if you modify the dictionary that you get back, then the next time you access the `definition`, you will get the modified version!

That's a problem *whenever* you use private attributes behind properties, which is a common idiom, so I'm not exactly worried about in the general case. But in the hash property, I delete the `'podpac_version'` key and all of the `'style'` keys from the definition. So I had to either downgrade the cached_property to a property or deepcopy the definition before deleting keys. Either seems fine, really.

(Since the hash and the json are still cached_properties, it seems like we won't hit the definition by itself all that often. But since the hash is also a cached_property, deep copying the definition once to produce it wouldn't be so bad either.)

Reply:
Switching to deepcopy